Removed Inline Styles For DevSite Examples

lokesh-0813 opened the pull request:
### Summary
Replaced Inline Style in JSX components for DEVSITE with SCSS files


resolves #498

StephenEsser commented:
There are a few other inline examples within other components. 

We released a new version of eslint-config-terra yesterday that warns when it detects an inline style. If you `npm run clean:install` you should get the updated package. Running `npm run lint` will log these warnings in the terminal. This should make finding them a little easier.

ShettyAkarsh commented on packages/terra-clinical-data-grid/src/terra-dev-site/doc/clinical-data-grid/DatagridExamples.2/DatagridWithCustomContent.jsx:
```diff
@@ -115,7 +115,7 @@ class DatagridWithCustomContent extends React.Component {
   render() {
     return (
-      <div style={{ height: '800px' }}>
+      <div className={cx('data-grid-div-style')}>
```
we can keep the name as 'data-grid-parent-div-style'

lokesh-0813 replied:
Classnames are renamed.

ShettyAkarsh commented on packages/terra-clinical-data-grid/src/terra-dev-site/doc/clinical-data-grid/DatagridExamples.2/DatagridWithSubsections.scss:
```diff
@@ -0,0 +1,5 @@
+:local {
+  .data-grid-div-style {
```
We can create a common css for this one. as the height value is common across most of the examples.

ShettyAkarsh commented on packages/terra-clinical-header/src/terra-dev-site/doc/example/HeaderLongTextWithContent.scss:
```diff
@@ -0,0 +1,5 @@
+:local {
```
even this one can be reused.

emilyrohrbough replied:
Lets create a common sass file to share across the examples and lets drop the @extends syntax

StephenEsser commented on packages/terra-clinical-data-grid/src/terra-dev-site/doc/clinical-data-grid/DatagridExamples.2/Datagrid.jsx:
```diff
@@ -74,7 +77,7 @@ class StaticDataGrid extends React.Component {
   render() {
     return (
-      <div style={{ height: '800px' }}>
+      <div className={cx('data-grid-div-style')}>
```
We should follow a similar approach to naming the example classes as we would for released components. I think we should avoid using the associate tag (`div`) and redundant (`style`) within the name. This class for example could be `data-grid-example` or something that describes what this example component is.

lokesh-0813 replied:
Classnames have been renamed with component names for better understanding.

lokesh-0813 commented:
> I think that this PR would be a lot cleaner if, especially for `DataGrid`, each test just referenced `_Datagrid.scss` and there was a single class with `height: 800px;` instead of each test having it's own `.scss` file with it's own single class in it.

@Blackop778 It was intentionally written in this approach to support any future modifications. For instance in `terra-grid` , there are scenarios where height is the only property but in few scenarios there is more than just height. Any upcoming modifications can be approached in ease.

StephenEsser commented:
I think the shared / extended SCSS for examples is becoming complicated. I'd like to get people's thoughts on removing any shared scss. Each example has its own scss file with no extended styles. 

Here is the primary thoughts I have on this:

We may eventually display the example scss to the user on the doc site. That may end up looking something like this:

![image](https://user-images.githubusercontent.com/6720991/58829462-778df680-860d-11e9-938d-9fd973e94995.png)

When the users clicks to display the scss with the current approach on this PR they would end up seeing something similar to this.

![image](https://user-images.githubusercontent.com/6720991/58829893-8f19af00-860e-11e9-8024-169839300e5d.png)

This makes the actual styles of the example abstracted away from the user and no longer provides much value.

But if the user was able to see the SCSS at face value they would be able to see the styles applied to the example:

![image](https://user-images.githubusercontent.com/6720991/58830054-08b19d00-860f-11e9-90b5-10a5c9145556.png)

emilyrohrbough commented:
I agree with @StephenEsser. Although `extends` is probably good practice in src code, I think using `extends` in our doc examples will make it harder for consumers to easily determine what styles where applied on a example.

StephenEsser commented:
Do you have any thoughts on this @bjankord ?

bjankord commented:
+1 to "removing any shared scss. Each example has its own scss file with no extended styles."
-1 to "We may eventually display the example scss to the user on the doc site." I don't believe that we'd want to display the styles in the doc site as we have been considering these implementation details. If people start to copy and paste styles out of terra components, those implementation details become more and more leaky.

StephenEsser commented:
The styles are shown to the user within the examples today with the use of inline styles.  My thoughts are that if the examples require css to display correctly, those implementation details should also be available to the user. Ideally the example code and the css to create that example are paired together. Copy pasting has always been pretty a tricky issue, I'll add this topic to the war room notes to talk through the pros and cons of hiding / showing the example styles a little more depth.

ShettyAkarsh commented:
@StephenEsser , @bjankord, @emilyrohrbough  should we go ahead and make this changes for the example files only ? by removing the shared scss  and keep the terra-dev-site->test folder shared scss implementation as its is ? for which we are not displaying the scss implementation for the user.

emilyrohrbough commented on packages/terra-clinical-data-grid/src/terra-dev-site/doc/clinical-data-grid/DatagridExamples.2/Datagrid.scss:
```diff
@@ -0,0 +1,8 @@
+
+:local {
+  .data-grid-basic {
+    @extend %data-grid-height;
```
Lets remove the @extend syntax and add the styles direclty.

emilyrohrbough commented on packages/terra-clinical-data-grid/src/terra-dev-site/doc/clinical-data-grid/DatagridExamples.2/Datagrid.scss:
```diff
@@ -0,0 +1,8 @@
+@import './Datagrid';
```
Lets drop the DataGrid import and add the styles directory or import both sass files into the example and class bind as needed

emilyrohrbough commented on packages/terra-clinical-data-grid/src/terra-dev-site/doc/clinical-data-grid/DatagridExamples.2/DatagridWithColumnResizing.scss:
```diff
@@ -0,0 +1,7 @@
+@import './Datagrid';
```
This is one-to-one styles to `Datagrid.scss`. Lets reuse `Datagrid.scss` and remove this file.

emilyrohrbough commented on packages/terra-clinical-data-grid/src/terra-dev-site/doc/clinical-data-grid/DatagridExamples.2/DatagridWithNoPinnedColumns.scss:
```diff
@@ -0,0 +1,7 @@
+@import './Datagrid';
```
This is one-to-one styles to `Datagrid.scss`. Lets reuse `Datagrid.scss` and remove this file.

emilyrohrbough commented on packages/terra-clinical-data-grid/src/terra-dev-site/doc/clinical-data-grid/DatagridExamples.2/DatagridWithSelections.scss:
```diff
@@ -0,0 +1,7 @@
+@import './Datagrid';
```
zThis is one-to-one styles to `Datagrid.scss`. Lets reuse `Datagrid.scss` and remove this file.

emilyrohrbough commented on packages/terra-clinical-data-grid/src/terra-dev-site/doc/clinical-data-grid/DatagridExamples.2/DatagridWithSubsections.scss:
```diff
@@ -0,0 +1,7 @@
+@import './Datagrid';
+
```
This is one-to-one styles to `Datagrid.scss`. Lets reuse `Datagrid.scss` and remove this file.

emilyrohrbough commented on packages/terra-clinical-data-grid/src/terra-dev-site/doc/clinical-data-grid/DatagridExamples.2/_Datagrid.scss:
```diff
@@ -0,0 +1,5 @@
+:local {
```
lets remove this file and add these styles directly

emilyrohrbough commented on packages/terra-clinical-data-grid/src/terra-dev-site/doc/clinical-data-grid/DatagridExamples.2/DatagridWithCustomContent.scss:
```diff
@@ -0,0 +1,12 @@
+@import './Datagrid';
```
IMO these docs styles can be merged into a common example with class names that are meaningful to their intent

emilyrohrbough commented on packages/terra-clinical-detail-view/src/terra-dev-site/doc/example/_Detailview.scss:
```diff
@@ -0,0 +1,5 @@
+:local {
```
Lets create a common sass file to share across the examples and lets drop the @extends syntax

lokesh-0813 commented:
@emilyrohrbough  @ShettyAkarsh @StephenEsser @Blackop778 
Updated examples to use styles from a common style file.
Removed `@extends` and `imports` implementation.

Also JEST test is failing for terra-onset-picker, hence updated OnsetPicker.test.jsx.snap.

emilyrohrbough commented on packages/terra-clinical-detail-view/src/terra-dev-site/doc/example/DetailViewDividedSmallerTitles.jsx:
```diff
@@ -23,7 +26,7 @@ const detail2item4 = (<LabelValueView label="Admin Notes" textValue="Try med for
 
 const DetailViewDivided = () => (
-  <div style={{ border: '3px dotted #d3d4d5' }}>
+  <div className={cx('detail-view-divided-small-title')}>
```
I think this class name needs updated to `detail-view-divided`

lokesh-0813 commented on packages/terra-clinical-detail-view/src/terra-dev-site/doc/example/DetailViewNoDivider.jsx:
```diff
@@ -2,14 +2,18 @@ import React from 'react';
 
 const DetailViewDivided = () => (
-  <div style={{ border: '3px dotted #d3d4d5' }}>
+  <div className={cx('detail-view-no-divider')}>
```
Yes @emilyrohrbough, I missed on that. I have them updated now. Thank you.

emilyrohrbough commented on packages/terra-clinical-data-grid/src/terra-dev-site/doc/clinical-data-grid/DatagridExamples.2/Datagrid.scss:
```diff
@@ -0,0 +1,20 @@
+:local {
```
@yuderekyu @bjankord shouldn't these sass files be named `name.module.scss`?

bjankord replied:
Left a comment on a similar PR in terra-core for this: https://github.com/cerner/terra-core/pull/2465#discussion_r295826939

lokesh-0813 replied:
Renamed name.scss to name.module.scss.

dkasper-was-taken commented on packages/terra-clinical-onset-picker/CHANGELOG.md:
```diff
@@ -3,6 +3,9 @@ ChangeLog
 
 Unreleased
 ----------
+### Changed
```
Is this change log entry needed, I don't see any corresponding changes. Otherwise looks good.
